mono/mini/interp/transform.c

```diff
@@ -5923,9 +5923,10 @@ mono_interp_transform_method (InterpMethod *imethod, ThreadContext *context, Mon
 	imethod = real_imethod;
 	mono_os_mutex_lock (&calc_section);
 	if (!imethod->transformed) {
-		InterpMethod *hash = imethod->next_jit_code_hash;
-		memcpy (imethod, &tmp_imethod, sizeof (InterpMethod));
-		imethod->next_jit_code_hash = hash;
+		// Ignore the first two fields which are unchanged. next_jit_code_hash shouldn't
+		// be modified because it is racy with internal hash table insert.
+		const int start_offset = 2 * sizeof (gpointer);
+		memcpy ((char*)imethod + start_offset, (char*)&tmp_imethod + start_offset, sizeof (InterpMethod) - start_offset);
```

Comment:
this looks a bit fragile. cant you do `&imethod->locals_size` / `&tmp_imethod->locals_size`?

Reply:
the idea was not to add dependency on the third field in the method, which is independent from the first two which are special